enhancements/sig-architecture/14-addon-cluster-proxy/README.md

```diff
@@ -0,0 +1,381 @@
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management/website/)
+
+## Summary
+
+Automating the installation of [apiserver-network-proxy](https://github.com/kubernetes-sigs/apiserver-network-proxy)
+so that admins or services can access the managed-clusters' kube-apiserver
+in an imperative approach but without retaining any per-cluster credentials
+in the hub cluster.
+
+## Motivation
+
+#### Mutual Cluster Connectivity
+
+Originally in OCM a minimal requirement for the underlying network 
+infrastructure is to support one-way network connectivity (spoke->hub) so 
+that the basic cluster registration and management can work. The
+apiserver-network-proxy will provide a reverse proxy tunnel over a long 
+connection and multiplexes packets through the tunnel so that hub and spoke 
+clusters are mutually accessible w/o raising more requirements to the 
+infrastructure. Note that the reverse tunnel is only supposed to transmit 
+"mouse flows" to avoid congestion.
+
+#### Direct Multi-Cluster Resource Access
+
+In some cases where we need to directly read the presence/state of specific 
+resources in the managed clusters, however for now we have no idea of 
+those resources unless they were created via the manifests api. Moreover,
+we don't have the fine-grained information about their fields and values 
+even if they are natively delivered by OCM. These problems will be eased
+in a degree if we can establish a proxy tunnel between hub and spoke 
+clusters.
+
+#### Avoid Potential Credential Leakage
+
+Normally we're supposed to obtain the credential (w/ sufficient permission)
+from the spoke cluster before raising api calls against the spoke cluster, 
+but that requires us to copy and dump the credential out of the spoke 
+cluster which exposes the cluster under potential danger of being accessed 
+by unexpected users. Also in some cases where the hub/spoke clusters are
+mutually untrusted, copying the credentials will be strictly impossible.
+
+
+### Goals
+
+Alpha: 
+
+- Automate the installation and operation and apiserver-network-proxy via 
+  the addon-framework.
+  - Structurelize the configuration of the proxy components from command-line
+    flags to a custom resource configuration object.
+- Automate the certificate rotation of both proxy-servers and proxy-agents.
+
+### Non-Goals
+
+- Tweak the original implementation of apiserver-network-proxy.
+
+### User Story
+
+#### 1. Calling spoke cluster's kubernetes api 
+
+## Proposal
+
+#### Architecture
+
+
+The following figure shows the interaction between the components:
+
+```
+   
+                     -----------------------------------------
+                     | Hub-Cluster                           |
+                     |                                       |
+  ----------------   | ----------------     ---------------  |
+  | Proxy Client |> > >| Proxy Server | <===| Proxy Addon |  |
```

Comment:
> There is a concern about proxy-server, if an agent register itself's name to proxy-server, and proxy-server cache this in memory. Then it would not be stateless.

sorry to be dense. can you elaborate what you mean by "stateless"?

Reply:
Sure, in our practice, we specify the number of instances for proxy-server to 1 because I thought the proxy-server cached some registration information.